core/templates/pages/story-editor-page/modal-templates/new-chapter-title-modal.controller.spec.ts

```diff
@@ -267,17 +271,36 @@ describe('Create New Chapter Modal Controller', function() {
     expect($uibModalInstance.close).not.toHaveBeenCalled();
   });
 
+  it('should prevent exploration from being added if its category ' +
+  'is not default', function() {
+    $scope.title = 'dummy_title';
+    var deferred = $q.defer();
+    deferred.resolve(true);
+    var deferred2 = $q.defer();
+    deferred2.resolve(false);
+    spyOn(explorationIdValidationService, 'isExpPublishedAsync')
+      .and.returnValue(deferred.promise);
+    spyOn(explorationIdValidationService, 'isCorrectnessFeedbackEnabled')
+      .and.returnValue(deferred.promise);
+    spyOn(explorationIdValidationService, 'isDefaultCategoryAsync')
+      .and.returnValue(deferred2.promise);
+    $scope.save();
+    $rootScope.$apply();
+    expect($scope.categoryIsDefault).toBe(false);
+    expect($uibModalInstance.close).not.toHaveBeenCalled();
```

Comment:
Please split this test into sections (setup, pre-check, action, post-cehck) by adding a newline

Reply:
Done!